pkg/services/ngalert/image/service.go

```diff
@@ -138,11 +141,32 @@ func (s *ScreenshotImageService) NewImage(ctx context.Context, r *models.AlertRu
 		OrgID:        r.OrgID,
 		DashboardUID: dashboardUID,
 		PanelID:      panelID,
+		From:         screenshot.DefaultFrom,
+		To:           screenshot.DefaultTo,
 		Timeout:      s.screenshotTimeout,
 	}
 
+	// Get the dashboard to set the time range for the screenshot
+	dashboard, err := s.dashboards.GetDashboard(ctx, &dashboards.GetDashboardQuery{
+		OrgID: r.OrgID,
+		UID:   dashboardUID,
+	})
+	if err != nil {
+		logger.Error("Failed to get dashboard for screenshot", "error", err)
+		return nil, err
+	}
+
+	tr, err := dashboard.GetTimeRange()
+	if err != nil {
+		logger.Warn("Failed to get time range from dashboard, using default",
+			"from", screenshot.DefaultFrom, "to", screenshot.DefaultTo)
+	} else {
+		opts.From = tr.From
+		opts.To = tr.To
+	}
+
```

Comment:
I feel like this might be better to do in `HeadlessScreenshotService.Take()` instead, what do you think? 

It would avoid a duplicate DB query, and this behavior feels like it isn't specific enough to not do in the general case (i.e. when that method is called without `To` and `From` being set in the options).

Reply:
Something like:

```go
func (s *HeadlessScreenshotService) Take(ctx context.Context, opts ScreenshotOptions) (*Screenshot, error) {
	// get dashboard

	if opts.To == "" && opts.From == "" {
		tr, err := dashboard.GetTimeRange()
		if err != nil {
			logger.Warn("Failed to get time range from dashboard, using default values")
		} else {
			opts.To = tr.To
			opts.From = tr.From
		}
	}

	opts = opts.SetDefaults()
	
	// ...
}
```

Reply:
Much better!

Reply:
There is an issue with this in that if the user changes the time range of the dashboard, we might use an older cached screenshot because the cache key is unchanged, but it's just for a short amount of time and probably not frequent enough to be a significant issue.

Reply:
That's a good point, we don't create a different key doing it that way. I agree it shouldn't be a significant issue but ideally the source of the hash would be unchanged after hashing. It might be worth documenting the behavior somewhere.